plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectPresenter.java

```diff
@@ -96,24 +95,34 @@ public void apply(List<ProjectDescriptor> projects) throws OperationException {
     }
 
 
-    /** {@inheritDoc} */
     @Override
     public boolean isCompleted() {
         if (view.isNewOpenShiftProjectSelected()) {
+            setUpNewProjectRequest();
+            setUpCheProjectRequest();
             return isOpenShiftProjectNameValid(view.getOpenShiftNewProjectName()) & isCheProjectNameValid(view.getCheNewProjectName());
         } else {
+            setUpExistProjectRequest();
+            setUpCheProjectRequest();
             return view.getExistedSelectedProject() != null && isCheProjectNameValid(view.getCheNewProjectName());
         }
     }
 
-    private boolean isOpenShiftProjectNameValid(String name) {
-        if (openShiftProjects.contains(name)) {
+    /**
+     * Checks whether project name matches OpenShift's naming requirements.
+     * If is not valid, display error message on the view, depending on case.
+     * @param projectName - project name
+     * @return true if project is valid, false otherwise
+     */
+    private boolean isOpenShiftProjectNameValid(String projectName) {
+        if (openShiftProjects.contains(projectName)) {
             view.showOsProjectNameError(locale.existingProjectNameError());
             return false;
         }
-        if (name.length() > 63
-            || name.isEmpty()
-            || !name.matches("[a-z0-9]([-a-z0-9]*[a-z0-9])?")) {
+        if (projectName.length() > 63
+            || projectName.length() < 2
+            || projectName.isEmpty()
+            || !projectName.matches("[a-z]([-a-z0-9]*[a-z0-9])?")) {
```

Comment:
"[a-z]([-a-z0-9]*[a-z0-9]" this line is used in two places. It would be better to make it as constant